### PR TITLE
Failure caused by use of stopping objects as if it was an object, not…

### PR DESCRIPTION
**Pull request by plwhite**
… a set.

stopping_objects_by_id is a dict whose values are sets, so treat as such.

### CODE REVIEW

**Comment by plwhite** [comment]
Simple fix, but really worth holding off until there are some tests for this code (and after feature/neutron-ha goes into master, at which point there may well be tests to extend).

**Comment by plwhite** [comment]
Fixes bug #541.

File: calico/felix/ipsets.py [hunk 1]
```diff
@@ -147,11 +147,15 @@ def cleanup(self):
...
                                                      n.startswith(tmppfx)])
+
+        # Ask the ipsets for all the names they may use and whitelist.
         whitelist = set()
-        for ipset in (self.objects_by_id.values() +
-                      self.stopping_objects_by_id.values()):
-            # Ask the ipset for all the names it may use and whitelist.
+        for ipset in (self.objects_by_id.values()):
+            whitelist.update(ipset.owned_ipset_names())
+        for ipset in [ipsets for ipsets in
```
Comment (fasaxc) [inline calico/felix/ipsets.py]: Isn't `[ii for ii in foo]` just an identity mapping?

Why don't I write some UT for the cleanup code and fix this as part of it?

Comment (fasaxc) [inline calico/felix/ipsets.py]: Never mind, brain fart on my part.  You're using update() on the line below.

**Comment by fasaxc** [comment]
I've put up a version with UT, which was on me anyway: #567

**Pull request closed without merging**